homeassistant/components/ws66i/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""The Soundavo WS66i 6-Zone Amplifier integration."""
+
+from pyws66i import get_ws66i
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_SOURCES, DOMAIN, INIT_OPTIONS_DEFAULT, WS66I_OBJECT
+
+PLATFORMS = ["media_player"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Soundavo WS66i 6-Zone Amplifier from a config entry."""
+    # Check if options was set. If not, initialize a dict for it
+    if not entry.options:
+        options = {CONF_SOURCES: INIT_OPTIONS_DEFAULT}
+        hass.config_entries.async_update_entry(entry, options=options)
+
+    entry.async_on_unload(entry.add_update_listener(_update_listener))
```

Comment:
Move this after the `ConfigEntryNotReady` so it doesn't have to get removed and recreated each time setup is retried